node/impl/full/mpool.go

```diff
@@ -108,7 +110,28 @@ func (a *MpoolAPI) MpoolPush(ctx context.Context, smsg *types.SignedMessage) (ci
 	return a.Mpool.Push(smsg)
 }
 
-func (a *MpoolAPI) MpoolPushMessage(ctx context.Context, msg *types.Message) (*types.SignedMessage, error) {
+func capGasFee(msg *types.Message, maxFee abi.TokenAmount) {
+	if maxFee.Equals(big.Zero()) {
+		return
+	}
+
+	chainFee := types.BigMul(msg.GasFeeCap, types.NewInt(uint64(msg.GasLimit)))
+	minerFee := types.BigMul(msg.GasPremium, types.NewInt(uint64(msg.GasLimit)))
+
+	if big.Add(chainFee, minerFee).LessThanEqual(maxFee) {
+		return
+	}
+
+	// scale chain/miner fee down proportionally to fit in our budget
+	// TODO: there are probably smarter things we can do here to optimize
+	//  message inclusion latency
+
+	totalFee := big.Add(chainFee, minerFee)
```

Comment:
@Kubuxu I may be confused here, is totalFee just GasFeeCap*GasLimit?

Reply:
Total max fee is `FeeCap * GasLimit` so yes.